more_itertools/more.py

```diff
@@ -545,3 +548,47 @@ def _get_values(self, value):
 
     def __getitem__(self, value):
         return self._get_values(value)
+
+
+def look_ahead(iterable, n=1):
+    """
+    Returns a 2-tuple with a list containing the first *n* elements of
+    *iterable*, and an iterator with the same items as *iterable*.
+    This allows you to "look ahead" at the items in the iterable without
+    advancing it.
+
+    There is one item in the list by default:
+
+        >>> iterable = 'abcdefg'
+        >>> head, iterable = look_ahead(iterable)
+        >>> head
+        ['a']
+        >>> list(iterable)
+        ['a', 'b', 'c', 'd', 'e', 'f', 'g']
+
+    You may use unpacking to retrieve items instead of lists:
+
+        >>> (head,), iterable = look_ahead('abcdefg')
+        >>> head
+        'a'
+        >>> (first, second), iterable = look_ahead('abcdefg', 2)
```

Comment:
I love this API: you get can your data singly or listly without needing to set an option.